packages/storage/e2e/StorageTask.e2e.js

```diff
@@ -578,9 +578,7 @@ describe('storage() -> StorageTask', () => {
       await promise;
     });
 
-    it('listens to download state', function testRunner() {
-      this.timeout(25000);
-
+    it('listens to download state', async () => {
```

Comment:
These storage test changes (and the related util change) are the things I'm really drooling over in this PR. The rest of the stuff is just trying to get it to a clean run so these can go in, I think? 

I'm working on the clean run as well in #4648 - once it lands I will try to integrate just the storage work maybe with an additional spy timeout as above and see how it goes. 


I've got #4648 clean on iOS just now, and both android and ios are more or less clean so that's my goal with this PR